hedera-node/src/main/java/com/hedera/services/store/contracts/HederaWorldState.java

```diff
@@ -109,57 +111,60 @@ public HederaWorldState(
 				provisionalAccount.setAutoRenew(sponsorMerkleAccount.getAutoRenewSecs());
 				provisionalAccount.setExpiry(sponsorMerkleAccount.getExpiry());
 			} else if (provisionalAccountUpdates.containsKey(sponsorAddress)) {
 				WorldStateAccount sponsorProvisionalAccount = provisionalAccountUpdates.get(sponsorAddress);
 				provisionalAccount.setProxyAccount(sponsorProvisionalAccount.getProxyAccount());
-				provisionalAccount.setKey(sponsorProvisionalAccount.getKey());
-				provisionalAccount.setMemo(sponsorProvisionalAccount.getMemo());
 				provisionalAccount.setAutoRenew(sponsorProvisionalAccount.getAutoRenew());
 				provisionalAccount.setExpiry(sponsorProvisionalAccount.getExpiry());
 			}
 		});
 
 		provisionalAccountUpdates.forEach((address, worldStateAccount) -> {
 			final var accountId = accountParsedFromSolidityAddress(address.toArray());
+			AccountState updatedAccount;
+
 			if (ledger.exists(accountId)) {
-				long oldBalance = ledger.getBalance(accountId);
-				long newBalance = worldStateAccount.getBalance().toLong();
-				long adjustment = newBalance - oldBalance;
-				ledger.adjustBalance(accountId, adjustment);
+				updatedAccount = repositoryRoot.getAccount(address.toArray());
 			} else {
-//				var key = new JContractIDKey(worldStateAccount.getProxyAccount().toGrpcContractId()); // TODO: might be used for factory contracts
-				HederaAccountCustomizer customizer = new HederaAccountCustomizer()
-						.key(worldStateAccount.getKey())
-						.memo("")
-						.proxy(worldStateAccount.getProxyAccount())
-						.expiry(worldStateAccount.getExpiry())
-						.autoRenewPeriod(worldStateAccount.getAutoRenew())
-						.isSmartContract(true);
+				updatedAccount = repositoryRoot.createAccount(address.toArray());
+				updatedAccount.setSmartContract(true);
+				updatedAccount.setExpirationTime(worldStateAccount.getExpiry());
+				updatedAccount.setAutoRenewPeriod(worldStateAccount.getAutoRenew());
+				updatedAccount.setProxyAccountShard(worldStateAccount.getProxyAccount().shard());
+				updatedAccount.setProxyAccountRealm(worldStateAccount.getProxyAccount().realm());
+				updatedAccount.setProxyAccountNum(worldStateAccount.getProxyAccount().num());
 				if (provisionalAccountDeletes.contains(address)) {
-					customizer.isDeleted(true);
+					updatedAccount.setDeleted(true);
 				}
-
-				long balance = worldStateAccount.getBalance().toLong();
-				ledger.spawn(accountId, balance, customizer);
 				createdContracts.add(EntityIdUtils.asContract(accountId));
 			}
+
+			updatedAccount.setBalance(worldStateAccount.getBalance().getAsBigInteger());
+			ledgerAccountsSource.put(address.toArray(), updatedAccount);
 		});
 
 		/* Commit code updates for each updated address */
 		provisionalCodeUpdates.forEach((address, code) -> repositoryRoot.saveCode(address.toArray(), code.toArray()));
 
 		repositoryRoot.flush();
 
+		if (creator.isPresent()) {
+			final var created = provisionalAccountUpdates.get(creator.get());
```

Comment:
So a directly created account gets key and memo, but second and third layer accounts don't?  This should be fine: `CREATE` gets no key and memo, but `ContractCreate` results do get it if I read it right.  A consistent division.

We cannot assume it's always the caller's info if we do add `CREATE`ed accounts.  A calls smart contract B that creates new contract C.  A is 'creator' but C would inherit B's meta-data.  A could do this call in initcode to be extra confusing.